components/skus/renderer/skus_render_frame_observer.cc

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/components/skus/renderer/skus_render_frame_observer.h"
+
+#include <string>
+#include <vector>
+
+#include "base/feature_list.h"
+#include "base/no_destructor.h"
+#include "brave/components/skus/common/features.h"
+#include "content/public/renderer/render_frame.h"
+#include "third_party/blink/public/platform/web_security_origin.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+
+namespace skus {
+
+SkusRenderFrameObserver::SkusRenderFrameObserver(
+    content::RenderFrame* render_frame,
+    int32_t world_id)
+    : RenderFrameObserver(render_frame), world_id_(world_id) {}
+
+SkusRenderFrameObserver::~SkusRenderFrameObserver() {}
+
+void SkusRenderFrameObserver::DidCreateScriptContext(
+    v8::Local<v8::Context> context,
+    int32_t world_id) {
+  if (!render_frame()->IsMainFrame() || world_id_ != world_id)
+    return;
+
+  if (!IsAllowed())
+    return;
+
+  if (!native_javascript_handle_) {
+    native_javascript_handle_ =
+        std::make_unique<SkusPageController>(render_frame());
+  } else {
+    native_javascript_handle_->ResetRemote(render_frame());
+  }
+
+  native_javascript_handle_->AddJavaScriptObjectToFrame(context);
+}
+
+bool SkusRenderFrameObserver::IsAllowed() {
+  DCHECK(base::FeatureList::IsEnabled(skus::features::kSkusFeature));
+  // NOTE: please open a security review when appending to this list.
+  static base::NoDestructor<std::vector<blink::WebSecurityOrigin>> safe_origins{
+      {{blink::WebSecurityOrigin::Create(GURL("https://account.brave.com"))},
```

Comment:
these are duplicating in several places and should be in common

Reply:
The only other place with them (currently) is in `brave/components/brave_vpn/brave_vpn_constants.h`. This would require some untangling so that it's in SKU instead and then VPN depends on SKU

I'd like to propose fixing this as a follow up (either in https://github.com/brave/brave-browser/issues/20374 or when evaluating desktop and what's left for VPN, as this PR doesn't finish VPN, it just provides SKU integration)